rest-api/src/main/resources/definitions/demographic_values_number_range_validation_rules.yml

```diff
@@ -0,0 +1,13 @@
+description: Validation for rules for validating numbers in demographics to ensure they fall within a specified range.
+type: object
+properties:
+  min:
+    description: The minimum allowed value for numbers in this category (optional).
+    type: number
+    format: double
```

Comment:
Will this use a double or a BigDecimal? If it uses a double, will this cause loss of precision?

Reply:
I was worried about that too so I looked into it before. It seems like [swagger itself doesn't specify a higher precision decimal type than double](https://swagger.io/docs/specification/data-models/data-types/#numbers). [It is possible to map every single double everywhere in the SDK to a BigDecimal](https://github.com/swagger-api/swagger-codegen/issues/5587#issuecomment-368805748). I didn't do this because BigDecimal operations are slower so that might have performance implications.

If someone needs additional precision they can still convert their decimal to a string, in which case it will not be touched at all.